### PR TITLE
Optimize SID translation

### PR DESCRIPTION
**Pull request by diversenok**
I suppose we can (at least partially) solve the performance issue reported in #349 by replacing multiple queries to LSA with one request to translate all SIDs at once.

### CODE REVIEW

File: phlib/lsasup.c [hunk 1]
```diff
@@ -269,6 +269,127 @@ NTSTATUS PhLookupSid(
...
+                    translatedNames[i] = PhConcatStrings(
+                        3,
+                        domainName->Buffer,
+                        L"\\",
+                        userName->Buffer
+                        );
+                }
+                else if (domainName)
+                {
+                    translatedNames[i] = PhDuplicateString(domainName);
```
Comment (dmex) [inline phlib/lsasup.c]: This should be either:
translatedNames[i] = PhReferenceObject(domainName);
or
PhSetReference(&translatedNames[i], domainName); // same as PhReferenceObject but inlined.
or 
PhSwapReference(&translatedNames[i], domainName);  // same as PhSetReference but inlined and frees any existing translatedNames entry.

File: phlib/lsasup.c [hunk 2]
```diff
@@ -269,6 +269,127 @@ NTSTATUS PhLookupSid(
...
+                }
+                else if (userName)
+                {
+                    translatedNames[i] = PhDuplicateString(userName);
```
Comment (dmex) [inline phlib/lsasup.c]: Same here

File: phlib/lsasup.c [hunk 3]
```diff
@@ -269,6 +269,127 @@ NTSTATUS PhLookupSid(
...
+                }
+            }
+            else
+            {
+                if (PhStartsWithString2(userName, L"S-1-", TRUE))
+                {
+                    translatedNames[i] = PhDuplicateString(userName);
```
Comment (dmex) [inline phlib/lsasup.c]: Here too

File: phlib/lsasup.c (hunk 1 above)
Comment (diversenok) [reply phlib/lsasup.c]: Sure.

**Comment by AltF5** [comment]
Oh wow cool, I’ll have to test this out!

**Pull request merged**